app/crud/sync_db/tile_cache_assets.py

```diff
@@ -38,10 +38,13 @@ def get_all_tile_caches():
                    WHERE assets.asset_type IN {str(tuple([e.value for e in list(TileCacheType)])).replace('"', "'")}
                     AND assets.status = 'saved'"""
         ).fetchall()
-    if not rows:
+    if rows is None:
         # tile_caches = []
-        logger.warning("There are no tile caches registered with the API.")
-    # else:
+        logger.warning("Rows is None. There are no tile caches registered with the API")
+    elif len(rows) == 0:
+        logger.warning("Zero rows. There are no tile caches registered with the API")
+    else:
+        logger.warning(f"Found {len(rows)} rows")
     # tile_caches = rows
```

Comment:
I'll admit having no tile caches is a rare situation (likely only in dev) but, in the interest of being thorough, let's clean up this bit which I vaguely recall we added while pairing. :)
The problem is that we check if rows is None (which it looks like it might be if there are no rows: Gino returns None, SQA would return an empty iterable IIRC), but then go on to do a for "for row in rows". If rows is None this will break because None is not iterable. So I think I would reduce this selection to logging a warning and then exiting early. For example:
if rows is None:
    logger.warning("There are no tile caches registered with the API")
    return []
for row in rows:
...

Reply:
Good catch, @dmannarino! 💪 
I hadn't looked at that code since we paired on it. 

I did refactor it. However, the function returns a `dict`. What I wanted to ensure is that `rows` are iterable as you suggest. Then the list comprehension never runs and the function returns the tile cache `dict` as before.

What do you think? My changes are in this commit: 8071e29a2b4c676031dc02a4cd750386ef0629dc

Reply:
Mmm, that looks good! I forgot the return value was a dictionary.